src/Types.hs

```diff
@@ -676,6 +677,19 @@ data Role =
 
 instance SOP.Generic Role
 
+data RoleScope
+  = ClassesScope (Set SchoolClass)
+  | SchoolScope
+  deriving (Eq, Ord, Show, Read, Generic)
+
+instance SOP.Generic RoleScope
+
+instance Monoid RoleScope where
```

Comment:
you should make a comment here and tell the story you told in the call this morning.  it's very insightful.